tuning/data/data_processors.py

```diff
@@ -452,9 +452,35 @@ def split_dataset(
         )
         return split_datasets
 
+    def _process_datasets_for_odm(
+        self,
+        processed_datasets: List[
+            Tuple[DataSetConfig, Union[DatasetDict, IterableDatasetDict]]
+        ],
+    ) -> Tuple[
+        Dict[str, Union[Dataset, IterableDataset]],
+        Dict[str, Union[Dataset, IterableDataset]],
+    ]:
+        train_split = "train"
+        eval_split = "test"
+        train_datasets_dict = {}
+        eval_datasets_dict = {}
+        for d, raw in processed_datasets:
+            if train_split in raw:
+                train_datasets_dict[d.name] = raw[train_split]
+            if eval_split in raw:
+                eval_datasets_dict[d.name] = raw[eval_split]
+        return train_datasets_dict, eval_datasets_dict
```

Comment:
Are we just looking to filter `train` and `test` splits from the dataset here?
Do you need the returned values to be a raw dict instead of a container?

Reply:
> Are we just looking to filter train and test splits from the dataset here?

Yes 

> Do you need the returned values to be a raw dict instead of a container?

dict would be good over datasetdict but both ways it should work.